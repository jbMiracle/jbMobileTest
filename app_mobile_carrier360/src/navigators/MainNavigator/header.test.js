import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MainNavHeader from './header';
import navigationOnHome from './navigation-snapshots/home.json';
import navigationOnOffers from './navigation-snapshots/offers.json';
import navigationInOfferDetails from './navigation-snapshots/offers.offer-details.json';

describe('MainNav header component', () => {
  it('should shallow render with the back button inactive on the Home tab', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<MainNavHeader navigation={ navigationOnHome } />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('should shallow render with the back button inactive on the Offers tab', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<MainNavHeader navigation={ navigationOnOffers } />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('should shallow render with the back button active when in Offer Details', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<MainNavHeader navigation={ navigationInOfferDetails } />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
