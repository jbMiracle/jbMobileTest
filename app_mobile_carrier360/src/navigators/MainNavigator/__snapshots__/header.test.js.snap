// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNav header component should shallow render with the back button active when in Offer Details 1`] = `
<Styled(C360Header)
  backButtonOnPress={[Function]}
  showBackButton={true}
  title="Offer Details"
  virtual={undefined}
/>
`;

exports[`MainNav header component should shallow render with the back button inactive on the Home tab 1`] = `
<Styled(C360Header)
  backButtonOnPress={[Function]}
  showBackButton={false}
  title="Back"
  virtual={undefined}
/>
`;

exports[`MainNav header component should shallow render with the back button inactive on the Offers tab 1`] = `
<Styled(C360Header)
  backButtonOnPress={[Function]}
  showBackButton={false}
  title="Back"
  virtual={undefined}
/>
`;
